examples/nested_parallel_memory.py

```diff
@@ -0,0 +1,124 @@
+"""
+==================================================
+Checkpoint using joblib.Memory and joblib.Parallel
+==================================================
+
+We illustrate how to cache intermediate computing results using
+:class:`joblib.Memory` within :class:`joblib.Parallel`.
+
+"""
+
+###############################################################################
+# Embed caching within parallel processing
+###############################################################################
+# 
+# It is possible to cache a computationally expensive function executing during
+# a parallel process. ``costly_column`` will emulate such function.
+
+import time
+
+
+def costly_column(data, column):
```

Comment:
This function name sounds like a noun, hence when reading the code, I don't have the impression that it is a function, but an array. Maybe I would call it "costly_compute"